src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -171,6 +218,61 @@ static unsigned uct_cuda_ipc_iface_progress(uct_iface_h tl_iface)
                                          max_events);
 }
 
+static ucs_status_t uct_cuda_ipc_iface_event_fd_arm(uct_iface_h tl_iface,
+                                                    unsigned events)
+{
+    uct_cuda_ipc_iface_t *iface = ucs_derived_of(tl_iface, uct_cuda_ipc_iface_t);
+    unsigned max_events         = iface->config.max_poll;
+    int ret;
+    int i;
+    uint64_t dummy;
+    ucs_status_t status;
+
+    if (uct_cuda_ipc_progress_event_q(iface, &iface->outstanding_d2d_event_q,
+                                      max_events)) {
```

Comment:
uct_cuda_ipc_progress_event_q() should not have max_events parameter, it should use iface->config.max_poll internally

Reply:
Agreed. @bureddy Can you look at the changes as well?